app/main.py

```diff
@@ -1,3 +1,67 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
```

Comment:
In this case, it's better to use `else` or `elif` because the app won't check other cases if the first condition is true.
```suggestion
        if isinstance(other, (int, float)):
            return Distance(self.km + other)
        else:
            return Distance(self.km + other.km)
```